libnetwork/netavark/config.go

```diff
@@ -230,8 +235,14 @@ func (n *netavarkNetwork) networkCreate(newNetwork *types.Network, defaultNet bo
 		}
 	}
 
-	// add gateway when not internal or dns enabled
-	addGateway := !newNetwork.Internal || newNetwork.DNSEnabled
+	// parse noAutoGateway
+	noAutoGateway, err := internalutil.ParseNoAutoGateway(newNetwork.Options[types.NoAutoGateway])
+	if err != nil {
+		return nil, err
+	}
+
+	// add gateway when not internal or dns enabled and no_auto_gateway not set
+	addGateway := (!newNetwork.Internal || newNetwork.DNSEnabled) && (!noAutoGateway)
```

Comment:
I just noticed but there is one problem with that, dns requires us to have a gateway ip so we can listen on it. Without it dns will not work. https://github.com/containers/common/commit/e030a76218e698f78bb9156b5c52f594f4c374d0

So we should either turn of DNS for this network if noAutoGw is set and throw a warning or just return an error and require users to create the bridge network with --disable-dns.

Reply:
This is an issue indeed. Somebody could add a gateway manually if he uses no_auto_gateway though. So I think we should warn/error if the final config does not contain a gateway but the network doesn't have --disable-dns set.

I'm not certain what all the --disable-dns does, so I'm gonna do some digging in the code. Wanne make sure that if there is a route to a local dns server this can still be utilized if set in /etc/resolv.conf without a default gateway.

Reply:
The solution now is to replace no_auto_gateway with the no_default_gateway option that prevents netavark from adding the default route but leaves the gateway in place. PR is containers/netavark#697

Reply:
Pushed the changes here for no_default_gateway